relational/table.py

```diff
@@ -230,58 +230,72 @@ def __init__(self):
         ])
 
     def _gen_facts(self, row, proj):
-        pts_pred = [('permit_type', lambda x: x == '1' or x == '2')]
-        if proj.field('address', PPTS.NAME) != '':
-            row[self.index(self.ADDRESS)] = proj.field('address', PPTS.NAME)
+        used_mohcd = False
+        for mohcd in _MOHCD_TYPES.keys():
+            if used_mohcd or proj.field('project_id', mohcd) == '':
+                continue
+
+            used_mohcd = True
+
+            name = proj.field('project_name', mohcd)
+
+            num = proj.field('street_number', mohcd)
+            addr = proj.field('street_name', mohcd)
+            if num:
+                addr = ('%s %s' % (num, addr))
+
+            if name:
+                addr = ('%s, %s' % (name, addr))
+
+            row[self.index(self.ADDRESS)] = '%s %s, %s' % (
+                    addr,
+                    proj.field('street_type', mohcd),
+                    proj.field('zip_code', mohcd))
+            sponsor = proj.field('project_lead_sponsor', mohcd)
+            if not sponsor:
+                sponsor = proj.field('project_sponsor', mohcd)
+            row[self.index(self.APPLICANT)] = sponsor
+
+            row[self.index(self.SUPERVISOR_DISTRICT)] = \
+                proj.field('supervisor_district', mohcd)
+
+            row[self.index(self.PERMIT_AUTHORITY_ID)] = proj.fk(mohcd)
+            row[self.index(self.PERMIT_AUTHORITY)] = 'mohcd'  # TODO
+
+        if used_mohcd:
+            return
```

Comment:
Did you move mohcd earlier becaues we trust their data more? If that's the case, should we also move PTS before PPTS?

Reply:
we use PTS as a fallback if we don't get information from PPTS, but we prioritize MOHCD _even if_ we have PPTS data. (all of this is only true for basic project facts.) so that's the distinction. i can add a comment to this effect. i also don't know if this is necessarily the best thing, but is what what "feels" right after playing with the data a bit.

Reply:
Gotcha, let's just add a comment or start adding stuff like this to Luke's doc